### PR TITLE
[OF-1475] fix: Updated Reparent Event Timing

### PR DESCRIPTION
**Pull request by MAG-mv**
- Reparent event now happens before the EntitySystem::DestroyEntity callback so clients can remove children before deletion
- Updated tests to ensure ordering

### CODE REVIEW

File: Library/src/Multiplayer/SpaceEntitySystem.cpp [hunk 1]
```diff
@@ -419,6 +409,25 @@ void SpaceEntitySystem::DestroyEntity(SpaceEntity* Entity, CallbackHandler Callb
 
 	CSP_DELETE(Keys);
 
+	csp::common::Array<ComponentUpdateInfo> Info;
+
+	RootHierarchyEntities.RemoveItem(Entity);
+
+	// Manually process the parent updates locally
+	// We want this callback to fire before the deletion so clients can react to children first
```
Comment (MAG-AdamThorn) [inline Library/src/Multiplayer/SpaceEntitySystem.cpp:417]: I think this comment needs to be updated as now happens after.

File: Library/src/Multiplayer/SpaceEntitySystem.cpp [hunk 2]
```diff
@@ -419,6 +409,25 @@ void SpaceEntitySystem::DestroyEntity(SpaceEntity* Entity, CallbackHandler Callb
...
+	auto ChildrenToUpdate = Children;
+
+	for (size_t i = 0; i < ChildrenToUpdate.Size(); ++i)
+	{
+		ChildrenToUpdate[i]->ParentId = nullptr;
```
Comment (MAG-OR) [inline Library/src/Multiplayer/SpaceEntitySystem.cpp:422]: [question] Is there a reason for not using RemoveParentEntity? (not a leading question at all, when I saw that I just thought there may be a function that does it already and I found that one)

**Review by MAG-ElliotMorris — APPROVED** [review]
LGTM

File: Library/src/Multiplayer/SpaceEntitySystem.cpp (hunk 2 above)
Comment (MAG-mv) [reply Library/src/Multiplayer/SpaceEntitySystem.cpp:422]: RemoveParentEntity is used when going through the patch process, as it also sets ShouldUpdateParent. We don't need to do that here, as we're updating manually

File: Library/src/Multiplayer/SpaceEntitySystem.cpp (hunk 1 above)
Comment (MAG-mv) [reply Library/src/Multiplayer/SpaceEntitySystem.cpp:417]: It still happens before. It's just outside of the callback now

**Review by MAG-OR — APPROVED** [review]
well done for adding tests

**Review by MAG-AdamThorn — APPROVED** [review]

**Pull request merged**